api/base/exceptions.py

```diff
@@ -45,3 +45,13 @@ class Gone(APIException):
 class InvalidFilterError(ParseError):
     """Raised when client passes an invalid filter in the querystring."""
     default_detail = 'Querystring contains an invalid filter.'
+
+
+class UnconfirmedAccountError(APIException):
+    status_code = 400
+    default_detail = 'Please confirm your account before using the API.'
+
+
+class DeactivatedAccountError(APIException):
+    status_code = 410
```

Comment:
This should be a 400 or 403.
410 should only  be regarding the resource you're attempting access.
> 403 Forbidden
>     The request was a valid request, but the server is refusing to respond to it. Unlike a 401 Unauthorized response, authenticating will make no difference.